src/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -81,13 +81,14 @@ public void SslStream_StreamToStream_DuplicateOptions_Throws()
                 serverOptions.ServerCertificate = certificate;
                 serverOptions.RemoteCertificateValidationCallback = AllowAnyServerCertificate;
 
-                Assert.ThrowsAsync<InvalidOperationException>(() => { return client.AuthenticateAsClientAsync(clientOptions, CancellationToken.None); });
-                Assert.ThrowsAsync<InvalidOperationException>(() => { return server.AuthenticateAsServerAsync(serverOptions, CancellationToken.None); });
+                Task t1 = Assert.ThrowsAsync<InvalidOperationException>(() => client.AuthenticateAsClientAsync(clientOptions, CancellationToken.None));
+                Task t2 = Assert.ThrowsAsync<InvalidOperationException>(() => server.AuthenticateAsServerAsync(serverOptions, CancellationToken.None));
+
+                await Task.WhenAll(t1, t2);
             }
```

Comment:
if t1, t2 don't complete this will result in hang no, should this be Assert.True(Task.WaitAll(t1,t2, timeout))

Reply:
no because then it will pass, as it won't wait for the asserts. There is a timeout already on the xunit tests, they don't hang for ever

Reply:
It wont pass if waitall completed becoz of timeout, as it will return false, and assert.true will throw.

Reply:
the only way to do this would be, 
Task delay = Task.Delay(sometimeout);
```
Task passingTask = Task.WhenAny(delay, Task.WhenAll(t1,t2));
Assert.NotEqual(passingTask, delay);
```

Reply:
Why wouldn't task.waitall with timeout work?

Reply:
Changed to wait all, seems odd still, you know that you can use a .json file in the solution folder to configure a timeout for all tests in a project rather than writing the code for each method.

Anyway updated to this as it seems the pattern currently used